stdlib/universe/typeconv.go

```diff
@@ -44,654 +43,298 @@ const (
-}
-func (c *stringConv) Bool() bool {
-	panic(values.UnexpectedKind(semantic.Function, semantic.Bool))
-}
-func (c *stringConv) Time() values.Time {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Time))
-}
-func (c *stringConv) Duration() values.Duration {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Duration))
-}
-func (c *stringConv) Regexp() *regexp.Regexp {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Regexp))
-}
-func (c *stringConv) Array() values.Array {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Array))
-}
-func (c *stringConv) Object() values.Object {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Object))
-}
-func (c *stringConv) Function() values.Function {
-	return c
-}
-func (c *stringConv) Dict() values.Dictionary {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Dictionary))
-}
-func (c *stringConv) Equal(rhs values.Value) bool {
-	f, ok := rhs.(*stringConv)
-	return ok && (c == f)
-}
-func (c *stringConv) HasSideEffect() bool {
-	return false
-}
-
-func (c *stringConv) Call(ctx context.Context, args values.Object) (values.Value, error) {
-	var str string
-	v, ok := args.Get(conversionArg)
-	if !ok {
-		return nil, errMissingArg
-	} else if v.IsNull() {
-		return values.Null, nil
-	}
-	switch v.Type().Nature() {
-	case semantic.String:
-		str = v.Str()
-	case semantic.Int:
-		str = strconv.FormatInt(v.Int(), 10)
-	case semantic.UInt:
-		str = strconv.FormatUint(v.UInt(), 10)
-	case semantic.Float:
-		str = strconv.FormatFloat(v.Float(), 'f', -1, 64)
-	case semantic.Bool:
-		str = strconv.FormatBool(v.Bool())
-	case semantic.Time:
-		str = v.Time().String()
-	case semantic.Duration:
-		str = v.Duration().String()
-	case semantic.Bytes:
-		var sb strings.Builder
-		var vB = v.Bytes()
-		for len(vB) > 0 {
-			r, size := utf8.DecodeRune(vB)
-			if r == utf8.RuneError && size == 1 {
-				return nil, errors.Newf(codes.Invalid, "invalid utf8 response")
+var stringConv = values.NewFunction(
+	"string",
+	convStringType,
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		var str string
+		v, ok := args.Get(conversionArg)
+		if !ok {
+			return nil, errMissingArg
+		} else if v.IsNull() {
+			return values.Null, nil
+		}
+		switch v.Type().Nature() {
+		case semantic.String:
+			str = v.Str()
+		case semantic.Int:
+			str = strconv.FormatInt(v.Int(), 10)
+		case semantic.UInt:
+			str = strconv.FormatUint(v.UInt(), 10)
+		case semantic.Float:
+			str = strconv.FormatFloat(v.Float(), 'f', -1, 64)
+		case semantic.Bool:
+			str = strconv.FormatBool(v.Bool())
+		case semantic.Time:
+			str = v.Time().String()
+		case semantic.Duration:
+			str = v.Duration().String()
+		case semantic.Bytes:
+			var sb strings.Builder
+			var vB = v.Bytes()
+			for len(vB) > 0 {
+				r, size := utf8.DecodeRune(vB)
+				if r == utf8.RuneError && size == 1 {
+					return nil, errors.Newf(codes.Invalid, "invalid utf8 response")
```

Comment:
nit, can you make this error message more clear to an end user who doesn't have the context of this code path?

i.e. something like `bytes contained non utf 8 bytes, cannot be converted into a string`